### DISCUSSION TITLE
Compiling from Milli to WebAssembly

### DISCUSSION THREAD

**Discussion by diegoleme** [meilisearch/product#409] [Feedback & Feature Proposal]
Is it possible to compile [milli](https://github.com/meilisearch/milli) for wasm and use it fully offline on a website?

**Comment by curquiza** [comment]
Hello @diegodem
This is not possible at the moment sorry!
Could you describe your use case and why you need this? 🙂

**Reply by gmourier** [reply]
Hey @curquiza and @diegoleme 👋

I checked with @Kerollmops today and it is supposed to be compilable for WASM. 

However, this won't help because the filesystem should be directly accessible from the portable generated binary for it to be usable and it seems not possible for security reasons I suppose.

It could be worth for to check  https://github.com/WICG/file-system-access/issues/139 since it mentions the use of a sandboxed file system but I doubt it's possible regarding lmdb (the backend storage Meilisearch use internally) and what it could imply in terms of performance.

**Reply by diegoleme** [reply]
The use case is to compile milli search and its index to wasm and be able to perform offline searches on a static site.

There are some tools emerging for this purpose, but not as mature as meilisearch.

- https://github.com/jameslittle230/stork
- https://github.com/tinysearch/tinysearch

**Reply by curquiza** [reply]
Thanks for your feedback @diegoleme! :)
